crypto/fipsmodule/sha/internal.h

```diff
@@ -28,14 +28,29 @@ extern "C" {
 #define SHA3_PAD_CHAR 0x06
 #define SHA3_ROWS 5
 
+// SHAKE constants, from NIST FIPS202.
+// https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf
+#define SHAKE_PAD_CHAR 0x1F
+#define SHAKE128_BLOCKSIZE (KECCAK1600_WIDTH - 128 * 2) / 8
+#define SHAKE256_BLOCKSIZE (KECCAK1600_WIDTH - 256 * 2) / 8
+
+// In the sponge construction, the width of the underlying 
+// function minus the rate (the block size in bits).
+// The |SHA3_MIN_CAPACITY_BYTES| corresponds to the 
+// lowest security level capacity for SHA3/SHAKE.
+// The data blocks increase when the capacity decreases.
+// The data buffer should have at least the maximum number of
+// block size bytes to fit any SHA3/SHAKE block length.
```

Comment:
these lines seem misplaced, so I would delete them from here.

Reply:
I just fixed that! Sorry.